src/command_modules/azure-cli-acr/azure/cli/command_modules/acr/_utils.py

```diff
@@ -91,6 +94,28 @@ def get_registry_by_name(cli_ctx, registry_name, resource_group_name=None):
     return client.get(resource_group_name, registry_name), resource_group_name
 
 
+def get_registry_from_name(cli_ctx, login_server=None, registry_name=None):
+    """Returns a Registry object for the specified name.
+    :param str name: either the registry name or the login server of the registry.
+    """
+    client = get_acr_service_client(cli_ctx).registries
+    registry_list = client.list()
+
+    if registry_name:
+        elements = [item for item in registry_list if
+                    item.login_server.lower() == login_server.lower() or item.name.lower() == registry_name.lower()]
```

Comment:
What if `login_server` is None?

Reply:
How does the caller even know from the method signature without reading its implementation?

Reply:
Line 53:
It promises a none empty value for source_registry because the value will be either from the command directly or from the interactivity by prompt. Again, as the client side, we cannot validate it however we can guarantee a nonempty value.

Similar for line 61 but the reason differs a little. If the source_registry_login_server is empty or none, an CLIError has already be thrown before reaching the call.

Reply:
But you are shipping an open source project with a util method that can possibly be used by anyone and it's clearly broken if called otherwise.

Reply:
The login_server should not be defaulted to None

Reply:
Basically validate all input especially if its exposed for other callers or handle null checks that can result in exceptions.

Reply:
For the login server, yes, caller is the layer to handle exceptions before the actual call.

Reply:
I think this method should be generic enough to accept either `name` or `login_server` (i.e., both should be default to None). Just use the non-empty ones to compare and filter the results.

I am hoping this method can further be reused to support scenarios such as `acr login -n registry.azurecr.io`.

Reply:
```
For the login server, yes, caller is the layer to handle exceptions before the actual call.
```
It makes no sense to me that your method takes a None value and doesn't handle it but throws a null reference exception out and wishes the caller can handle it.

Reply:
I mean the caller throws an exception when the login server is empty before the method actually get called. Please see Line 59 for now. Btw I will combine the two errors together as you suggested : )